packages/ketcher-core/src/application/editor/modes/SequenceMode.ts

```diff
@@ -835,20 +835,51 @@ export class SequenceMode extends BaseMode {
     const chainsCollection = ChainsCollection.fromMonomers([
       ...drawingEntitiesManager.monomers.values(),
     ]);
+
+    const currentSequence = SequenceRenderer.currentChain;
+
+    const currentSequenceHasPhosphate = currentSequence?.lastNonEmptyNode
+      ? this.hasPhosphateAtEndOfSequence(
+          currentSequence.lastNonEmptyNode.monomer,
+        )
+      : false;
+
+    const newSequenceHasPhosphate = chainsCollection.lastNode
+      ? this.hasPhosphateAtEndOfSequence(chainsCollection.lastNode.monomer)
+      : false;
+
+    let nextCaretPosition =
+      SequenceRenderer.caretPosition + chainsCollection.length;
+
+    if (
+      (currentSequenceHasPhosphate && newSequenceHasPhosphate) ||
+      currentSequenceHasPhosphate
+    ) {
+      nextCaretPosition -= 1;
+    }
+
     const modelChanges = this.insertNewSequenceFragment(chainsCollection);
 
     modelChanges.addOperation(new ReinitializeModeOperation());
 
     modelChanges.addOperation(
       new RestoreSequenceCaretPositionOperation(
         SequenceRenderer.caretPosition,
-        SequenceRenderer.caretPosition + chainsCollection.length,
+        nextCaretPosition,
       ),
     );
 
     return modelChanges;
   }
 
+  private hasPhosphateAtEndOfSequence(monomer: BaseMonomer): boolean {
```

Comment:
Could you please elaborate why exactly this fix works? For me it seems that you're targeting some edge case here rather than addressing the general issue, but I may be wrong

Reply:
The bug was due to incorrect calculation of the caret.


We solve, in fact, a particular problem of the position of the caret, if at the end of the sequence there is a Phosphate residue (p), then when inserting another sequence, this residual is transferred to the end of the resulting sequence. 


And we need to take into account several variants of events, whether there is a “p” on an existing one or on two simultaneous sequences. 

If the condition is true, we decrement the caret position by 1. Since "p" does not "add up". 

e.g. `AAAp + CCCp = AAACCCp (not AAACCCpp)`

But if “p” is only on the new sequence, then we do not change the position of the caret.

Therefore, we use the `hasPhosphateAtEndOfSequence` method to check at the end of the sequence. =)